contents/android/IntroToAndroid.md

```diff
@@ -0,0 +1,108 @@
+
+#### Services
+A `Service` performs long-running operations in the foreground or background, and it does not provide a user interface. For example, a `Service` can play audio in the foreground or download data in the background. Read more at the [Services](https://developer.android.com/guide/components/services.html) page in the Android Developer Guides.
+
+#### Broadcast receivers
+A `Broadcast Receiver` allows the app to register for events sent by the System, events sent by another application or events sent within the same application. Even if the app is inactive, the app is able to respond to system-wide broadcast announcements. Read more at the [Broadcasts](https://developer.android.com/guide/components/broadcasts.html) page.
+
+#### Content providers
+A `Content provider` abstracts the data layer, encapsulating data and allowing for data security. Read more at the [Content providers](https://developer.android.com/guide/topics/providers/content-providers.html) page.
+
+### 2.3 Activity Lifecycle
+![Activity Lifecycle](https://developer.android.com/guide/components/images/activity_lifecycle.png)
+
+([Image](https://developer.android.com/guide/components/images/activity_lifecycle.png) from Google)
+
+Activity instances in an app will transition through different stages in their `lifecycle`. An activity that has just been created will be in the `Created` stage. When the app prepares for the activity to enter the foreground and be displayed to the user, it enters the `Started` stage. The UI could be initialised at this stage. When the activity comes to the foreground and can be viewed by the user, it enters the `Resumed` stage. More details about the other stages can be found in the picture above or at the [Activity Lifecycle](https://developer.android.com/guide/components/activities/activity-lifecycle.html) page.
+
+The `Activity` class implements callbacks that will be invoked as the activity enters a new state. These callbacks can be overridden in your activity class that inherits from `Activity` so that your app can respond to transitions between states.
+
+The following code is an example of overriding the `onCreate` callback. It calls the superclass method first and then declares the user interface which is defined in an XML layout file `R.layout.activity_main`.
+
+```
+public class MainActivity extends Activity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+    }
+```
+
```

Comment:
Not sure if you want to talk about resource management & how the activity lifecycle relates. A quick mention might be nice.